app/src/main/java/com/graeberj/taskman/agenda/data/mapper/AgendaMapper.kt

```diff
@@ -99,3 +101,18 @@ fun AgendaItem.Task.toTaskDto(): TaskDto {
     )
 }
 
+fun AgendaResponseDto.toAgendaItemsList(): List<AgendaItem> {
+    val agendaItems = mutableListOf<AgendaItem>()
+    events.forEach { event ->
+        agendaItems.add(event.toEvent())
+    }
+    reminders.forEach { reminder ->
+        agendaItems.add(reminder.toReminder())
+    }
+    tasks.forEach { task ->
+        agendaItems.add(task.toTask())
+    }
+
+    return agendaItems
+}
```

Comment:
You could use map here to avoid needing the mutable list: 
```
val events = this.events.map { it.toEvent() }
val reminders = this.reminders.map { it.toReminder() }
val tasks = this.tasks.map { it.toTask() }

return events + reminders + tasks
```